poly/src/evaluations.rs

```diff
@@ -1,4 +1,4 @@
-//! A polynomial represented in evaluations form.
+//! A univariate dense polynomial represented in evaluations form.
```

Comment:
```suggestion
//! A univariate polynomial represented in evaluations form.
```

Reply:
Any thoughts on how would the API need to be changed to represent a multilinear polynomial in evaluation form? (Not needed for this PR)

Reply:
Not sure if it is the best solution, but my current idea is to let multilinear polynomial implement `MVPolynomial`, and `MVPolynomial` no longer has method `terms`, `from_coefficients...`. Those methods are moved to another trait called `MVPolynomialCoefficientForm` that extends `MVPolynomial`. Ryan's `SparsePolynomial` implements this trait. Similarly, there is a trait called `MultilinearPolynomialEvaluationForm` that also extends `MVPolynomial`. In this PR, I implemented `partial_evaluate`, which reduces the number of variables by evaluating part of the vector. There's also a handy method called `relabel`, which basically switches positions of scalars in a vector (by doing so, it is easy to change the order of argument of the polynomial). 

For future PRs, I'm thinking about whether we should let univariate evaluations implement `UVPolynomial` trait.